gcloud-java-storage/src/main/java/com/google/gcloud/spi/DefaultStorageRpc.java

```diff
@@ -415,7 +415,7 @@ public void onFailure(GoogleJsonError e, HttpHeaders responseHeaders) {
           .setIfGenerationMatch(IF_GENERATION_MATCH.getLong(options))
           .setIfGenerationNotMatch(IF_GENERATION_NOT_MATCH.getLong(options));
       MediaHttpDownloader downloader = req.getMediaHttpDownloader();
-      downloader.setContentRange(position, (int) position + bytes);
+      downloader.setContentRange(position, (int) position + bytes - 1);
```

Comment:
Shall we do:
[Ints.checkedCast](http://docs.guava-libraries.googlecode.com/git/javadoc/src-html/com/google/common/primitives/Ints.html#line.87)(position + bytes - 1) until the http-api-client bug is fixed?